app/database/components/index.tsx

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ComponentType, useEffect, useState} from 'react';
+import {Database} from '@nozbe/watermelondb';
+import DatabaseProvider from '@nozbe/watermelondb/DatabaseProvider';
+
+import {MM_TABLES} from '@constants/database';
+import DatabaseManager from '@database/manager';
+import Servers from '@database/models/default/servers';
+
+const {SERVERS} = MM_TABLES.DEFAULT;
+
+//TODO: Remove when DatabaseManager is Singleton
+const DBManager = new DatabaseManager();
+
+export function withServerDatabase<T>(
+    Component: ComponentType<T>,
+): ComponentType<T> {
+    return function ServerDatabaseComponent(props) {
+        const [database, setDatabase] = useState<Database|unknown>();
+
+        // If we don't need to await the async functions this side effect is not needed
+        useEffect(() => {
+            const observer = async (servers: Servers[]) => {
+                const server = servers.reduce((a, b) => (a.lastActiveAt > b.lastActiveAt ? a : b));
+
+                // The server database should already exists at this point
+                // there should not be a need to await
+                const serverDatabase = await DBManager.retrieveDatabaseInstances([server.url]);
```

Comment:
1) Since we now have the `last_active_at` column, we will need to update the methods inside `DatabaseManager` for them to read this value instead of the one present under GLOBAL entity with name `RECENTLY_VIEWED_SERVERS`. 

2) If we don't do part (1) in this PR, then we can update the code here to use the changes https://github.com/mattermost/mattermost-mobile/commit/ee9fffc3d0bbb76ae4b2e7738af16b8d05040742

Reply:
I'm happy to help with (1) and (2).  Just let me know @enahum